DXCDT-138: "Using as a CLI" doc page [2/10] (#568)
auth0/auth0-deploy-cli: docs/using-as-cli.md

> # Using as a CLI

Author: Resolved because it's been added with #578

> The Deploy CLI, can be used as a standalone command line utility. Doing so provides a simple way to manage your Auth0 tenant configuration in CI/CD workflows.

Reviewer: ```suggestion
The Deploy CLI can be used as a standalone command line utility. Doing so provides a simple way to manage your Auth0 tenant configuration in CI/CD workflows.
```

## `export` command

Fetching configurations from Auth0 tenant to the local machine.

### `--output_folder`, `-o`

Path. Specifies the target directory for configuration files to be written to.

### `--config_file`, `-c`

Path. Specifies the user-defined configuration file (`config.json`). Refer to the list of [all configurable properties](#).

### `--format`, `-f`

> Options: `yaml` or `directory`. Determines the Refer to the list of all configurable properties.

Reviewer: > Determines the Refer to the list

This seems to be incomplete.

> Options: yaml or directory. Determines the Refer to the list of all configurable properties.

Reviewer 2: ```suggestion
Options: **yaml** or **directory**. Determines the Refer to the list of all configurable properties.
```

Just a suggestion to make it stand out more.

### `--export_ids`, `-e`

> Boolean: When enabled, will export the identifier fields for each resource. Default: `false`.

Reviewer 2: ```suggestion
Boolean. When enabled, will export the identifier fields for each resource. Default: `false`.
```

### `--env`

> Boolean. Indicates if tools should ingest environment variables or not. Default: `true`.

Reviewer: ```suggestion
Boolean. Indicates if the tool should ingest environment variables or not. Default: `true`.
```

### `--debug`

Boolean. Enables more verbose error logging; useful during troubleshooting. Default: `false`.

### `--proxy_url`, `-p`

> A url for proxying requests, only set this if you are behind a proxy.

Reviewer: ```suggestion
A url for proxying requests. Only set this if you are behind a proxy.
```

### Examples

```shell
# Fetching Auth0 tenant configuration in the YAML format
a0deploy export -c=config.json --format=yaml --output_folder=local

# Fetching Auth0 tenant configuration in directory (JSON) format
a0deploy export -c=config.json --format=directory --output_folder=local

# Fetching Auth0 tenant configurations with IDs of all assets
a0deploy export -c=config.json --format=yaml --output_folder=local --export_ids=true
```

## `import` command

Applying configurations from local machine to Auth0 tenant.

### `--input_file`, `-i`

Path. Specifies the location of the resource configuration files. For YAML formats, this will point to the `tenant.yaml` file, for directory formats, this will point to the resource configuration directory.

### `--config_file`, `-c`

Path. Specifies the user-defined configuration file (config.json). Refer to the list of [all configurable properties](#).

### `--env`

> Boolean. Indicates if tools should ingest environment variables or not. Default: `true`.

Reviewer: ```suggestion
Boolean. Indicates if the tool should ingest environment variables or not. Default: `true`.
```

### `--proxy_url`, `-p`

> A url for proxying requests, only set this if you are behind a proxy.

Reviewer: ```suggestion
A url for proxying requests. Only set this if you are behind a proxy.
```
